### PR TITLE
expression: implement vectorized evaluation for `builtinDecimalAnyValueSig`

### PR DESCRIPTION
**Pull request by KuangBo**
### What problem does this PR solve? 
expression: implement vectorized evaluation for `builtinDecimalAnyValueSig`

### What is changed and how it works?
```
go test -v -benchmem -bench=BenchmarkVectorizedBuiltinMiscellaneousFunc -run=BenchmarkVectorizedBuiltinMiscellaneousFunc -args builtinDecimalAnyValueSig
goos: linux
goarch: amd64
pkg: github.com/pingcap/tidb/expression
BenchmarkVectorizedBuiltinMiscellaneousFunc/builtinDecimalAnyValueSig-VecBuiltinFunc-4         	  755986	      1466 ns/op	       0 B/op	       0 allocs/op
BenchmarkVectorizedBuiltinMiscellaneousFunc/builtinDecimalAnyValueSig-NonVecBuiltinFunc-4      	   53552	     21861 ns/op	       0 B/op	       0 allocs/op
PASS
ok  	github.com/pingcap/tidb/expression	2.545s
```

### Check List 

Tests 

 - Unit test

### CODE REVIEW

**Comment by KuangBo** [comment]
> Thanks for your contribution~
> Your benchmark result seems wrong, which should contain a row-based function's result and a vectorized function's result.
> Please run `go test -v -bench=TestVectorizedBuiltinMiscellaneousCasesFunc -run=TestVectorizedBuiltinMiscellaneousCasesFunc -args builtinDecimalAnyValueSig` under the expression package.

Thanks for your advise.But my results are as follows:
```
go test -v -bench=TestVectorizedBuiltinMiscellaneousCasesFunc -run=TestVectorizedBuiltinMiscellaneousCasesFunc -args builtinDecimalAnyValueSig
PASS
ok  	github.com/pingcap/tidb/expression	0.062s
```
Is that right？

**Comment by KuangBo** [comment]
### What problem does this PR solve?
expression: implement vectorized evaluation for builtinMicroSecondSig

### What is changed and how it works?
```
goos: linux
goarch: amd64
pkg: github.com/pingcap/tidb/expression
BenchmarkVectorizedBuiltinTimeFuncGenerated-4   	1000000000	         0.0101 ns/op	       0 B/op	       0 allocs/op
BenchmarkVectorizedBuiltinTimeFunc/builtinMicroSecondSig-VecBuiltinFunc-4         	  342348	      3660 ns/op	       0 B/op	       0 allocs/op
BenchmarkVectorizedBuiltinTimeFunc/builtinMicroSecondSig-NonVecBuiltinFunc-4      	   38652	     30583 ns/op	       0 B/op	       0 allocs/op
PASS
ok  	github.com/pingcap/tidb/expression	2.915s
```
### Check List
Tests

- Unit test

**Comment by KuangBo** [comment]
@qw4990 Excuse me, could you come over and look look this PR?  Another commit in the queue. Thanks!

**Comment by SunRunAway** [comment]
The name of test functions and benchmark functions in builtin_miscellaneous_vec_test.go is not well formatted. Please help to correct it.

`TestVectorizedBuiltinMiscellaneousCasesEvalOneVec` -> `TestVectorizedBuiltinMiscellaneousEvalOneVec`

`TestVectorizedBuiltinMiscellaneousCasesFunc` -> `TestVectorizedBuiltinMiscellaneousFunc`
`BenchmarkVectorizedBuiltinMiscellaneousCasesEvalOneVec` -> `BenchmarkVectorizedBuiltinMiscellaneousEvalOneVec`
`BenchmarkVectorizedBuiltinMiscellaneousCasesFunc` -> `BenchmarkVectorizedBuiltinMiscellaneousFunc`

**Comment by KuangBo** [comment]
@SunRunAway  Thanks. My results are as follows:
```
go test -v -benchmem -bench=BenchmarkVectorizedBuiltinMiscellaneousFunc -run=BenchmarkVectorizedBuiltinMiscellaneousFunc -args builtinDecimalAnyValueSig
goos: linux
goarch: amd64
pkg: github.com/pingcap/tidb/expression
BenchmarkVectorizedBuiltinMiscellaneousFunc/builtinDecimalAnyValueSig-VecBuiltinFunc-4         	  755986	      1466 ns/op	       0 B/op	       0 allocs/op
BenchmarkVectorizedBuiltinMiscellaneousFunc/builtinDecimalAnyValueSig-NonVecBuiltinFunc-4      	   53552	     21861 ns/op	       0 B/op	       0 allocs/op
PASS
ok  	github.com/pingcap/tidb/expression	2.545s
```
```
go test -v -benchmem -bench=BenchmarkVectorizedBuiltinMiscellaneousCasesFunc -run=BenchmarkVectorizedBuiltinMiscellaneousCasesFunc -args builtinDecimalAnyValueSig
goos: linux
goarch: amd64
pkg: github.com/pingcap/tidb/expression
BenchmarkVectorizedBuiltinMiscellaneousCasesFunc/builtinDecimalAnyValueSig-VecBuiltinFunc-4         	  831997	      1437 ns/op	       0 B/op	       0 allocs/op
BenchmarkVectorizedBuiltinMiscellaneousCasesFunc/builtinDecimalAnyValueSig-NonVecBuiltinFunc-4      	   54309	     21825 ns/op	       0 B/op	       0 allocs/op
PASS
ok  	github.com/pingcap/tidb/expression	3.600s
```
Do I need to submit the modified code？

**Comment by KuangBo** [comment]
@SunRunAway I have commited the code. Please review.

File: expression/builtin_miscellaneous_vec_test.go [hunk 1]
```diff
@@ -81,3 +82,19 @@ func BenchmarkVectorizedBuiltinMiscellaneousCasesEvalOneVec(b *testing.B) {
 func BenchmarkVectorizedBuiltinMiscellaneousCasesFunc(b *testing.B) {
 	benchmarkVectorizedBuiltinFunc(b, vecBuiltinMiscellaneousCases)
 }
+
```
Comment (qw4990) [inline expression/builtin_miscellaneous_vec_test.go]: And please help us to remove these four methods above here(line 70~84);

Comment (KuangBo) [reply expression/builtin_miscellaneous_vec_test.go]: OK, I have removed.

**Comment by KuangBo** [comment]
@SunRunAway Hello, please help me ^_^

**Comment by SunRunAway** [comment]
Hi, @KuangBo 
You can post your benchmark result into the PR description, Thanks.

**Comment by KuangBo** [comment]
Oh, thank you @SunRunAway 
This is my negligence, I have modified it.

**Pull request merged**